src/js/components/componentDiagram/index.js

```diff
@@ -327,6 +327,57 @@ function renderGraph(componentDiagram) {
   componentDiagram.emit('postrender');
 }
 
+function scrollToNodes(componentDiagram, nodes) {
+  const nodesBox = {
+    top: [],
+    left: [],
+    right: [],
+    bottom: [],
+    x: [],
+    y: [],
+  };
+
+  nodes.forEach((id) => {
+    const node = componentDiagram.graph.node(id);
+    if (!node) {
+      return;
+    }
+
+    const nodeBox = node.elem.getBoundingClientRect();
+    nodesBox.top.push(nodeBox.top);
+    nodesBox.left.push(nodeBox.left);
+    nodesBox.right.push(nodeBox.right);
+    nodesBox.bottom.push(nodeBox.bottom);
+    nodesBox.x.push(node.x - nodeBox.width / 2);
+    nodesBox.y.push(node.y - nodeBox.height / 2);
+  });
+
+  nodesBox.top = Math.min(...nodesBox.top);
+  nodesBox.left = Math.min(...nodesBox.left);
+  nodesBox.right = Math.max(...nodesBox.right);
+  nodesBox.bottom = Math.max(...nodesBox.bottom);
+  nodesBox.offsetTop = Math.min(...nodesBox.y);
+  nodesBox.offsetLeft = Math.min(...nodesBox.x);
+
+  nodesBox.width = nodesBox.right - nodesBox.left;
+  nodesBox.height = nodesBox.bottom - nodesBox.top;
+
+  const { containerController } = componentDiagram.container;
+  const containerBox = containerController.element.getBoundingClientRect();
+
+  const xRatio = containerBox.width / nodesBox.width;
+  const yRatio = containerBox.height / nodesBox.height;
+  const scale = (xRatio > 1 && yRatio > 1) ? 1 : Math.min(xRatio, yRatio) - 0.01;
+
+  containerController.scaleTo(scale);
+
+  setTimeout(() => {
+    const x = nodesBox.width / 2 + nodesBox.offsetLeft;
+    const y = nodesBox.height / 2 + nodesBox.offsetTop;
+    containerController.translateTo(x, y);
```

Comment:
If the node is already visible, is it going to move it again? Because we probably don't want that. Otherwise, when a user clicks on a node to highlight it, the node will move. Right?

Reply:
Yes, I assume node should be centered on the graph every time when `scrollTo` was called.
Doesn't matter it's visible or not.

Reply:
Ok but, if the user is clicking on the node, we won’t want it to move if it’s already visible.